site/ja/guide/create_op.md

```diff
@@ -0,0 +1,1231 @@
+  //
+  // 計算のためのblock数とthread_per_block数の例は、
+  // core/util/gpu_kernel_helper.hを見てください。
+  int block_count = 1024;
+  int thread_per_block = 20;
+  ExampleCudaKernel<T>
+      <<<block_count, thread_per_block, 0, d.stream()>>>(size, in, out);
+}
+
+// 登録されたOpKernelの型のために、明示的にファンクタをインスタンス化します。
+template struct ExampleFunctor<GPUDevice, float>;
+template struct ExampleFunctor<GPUDevice, int32>;
+
+#endif  // GOOGLE_CUDA
+```
+
+
+## オペレーションライブラリの構築
+
+### システムのコンパイラを使ってコンパイル（TensorFlowのバイナリインストール）
+
+システム上で利用できる `g++` や `clang` のような `C++` コンパイラを使って、`zero_out.cc` をコンパイルするはずです。
+バイナリのPIPパッケージは、オペレーションをコンパイルするために必要な、ヘッダファイルとライブラリをシステム固有の場所にインストールします。
+しかし、TensorFlowのpythonライブラリは、ヘッダのディレクトリを取得する `get_include` 関数と、リンクされる共有オブジェクトがあるディレクトリを取得する `get_lib` 関数を提供しています。
+Ubuntuマシン上におけるこれらの関数の出力を、次に示します。
+
+```bash
+$ python
+>>> import tensorflow as tf
+>>> tf.sysconfig.get_include()
+'/usr/local/lib/python2.7/site-packages/tensorflow/include'
+>>> tf.sysconfig.get_lib()
+'/usr/local/lib/python2.7/site-packages/tensorflow'
+```
+
+`g++` がインストールされていることを想定し、ここではオペレーションを動的ライブラリにコンパイルするための一連のコマンドを示します。
+
+```bash
+TF_CFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_compile_flags()))') )
+TF_LFLAGS=( $(python -c 'import tensorflow as tf; print(" ".join(tf.sysconfig.get_link_flags()))') )
+g++ -std=c++11 -shared zero_out.cc -o zero_out.so -fPIC ${TF_CFLAGS[@]} ${TF_LFLAGS[@]} -O2
+```
+
+Mac OS XX上では、`.so` ファイルをビルドするときに、追加フラグ "-undefined dynamic_lookup" が必要です。
+
+> `gcc` のバージョンが `>=5` のときの注意点: gccは、バージョン `5` から新しいC++ [ABI](https://gcc.gnu.org/gcc-5/changes.html#libstdcxx) を利用します。TensorFlowのウェブサイトで利用可能なパッケージは、古いABIを利用する `gcc4` でビルドされています。もしオペレーションを `gcc>=5` でコンパイルする場合、コマンドラインに `-D_GLIBCXX_USE_CXX11_ABI=0` を追加し、古いABIと互換をもたせるようにしてください。
+
+
+### bazelを使ってオペレーションをコンパイル（TensorFlowのソースコードインストール）
+
+もしTensorFlowのソースコードがインストールされているなら、オペレーションをコンパイルするためにTensorFlowのビルドシステムを使用できます。
+Bazelのビルドルールに従ったBUILDファイルを [`tensorflow/core/user_ops`][user_ops] ディレクトリに配置してください。
+
+```python
+load("//tensorflow:tensorflow.bzl", "tf_custom_op_library")
+
+tf_custom_op_library(
+    name = "zero_out.so",
+    srcs = ["zero_out.cc"],
+)
+```
+
+`zero_out.so` をビルドするために、次のコマンドを実行します。
+
+```bash
+$ bazel build --config opt //tensorflow/core/user_ops:zero_out.so
+```
+
+> 前述したように、もしgcc>=5でコンパイルする場合は、bazelのコマンドラインに `--cxxopt="-D_GLIBCXX_USE_CXX11_ABI=0"` を追加してください。
+
+> Note: 標準の `cc_library` ルールを利用して共有ライブラリ（`.so` ファイル）を作成できますが、`tf_custom_op_library` マクロを利用することを強く推奨します。このマクロは、必要となるいくつかの依存関係を追加し、共有ライブラリがTensorFlowのプラグイン読み込み機構と適合しているかを確認します。
+
+
+## オペレーションをPythonで使用する
+
+TensorFlowのPython APIは、動的ライブラリをロードしてTensorFlowのフレームワークにオペレーションを登録する、`tf.load_op_library` 関数を提供しています。
+`load_op_library` は、オペレーションとカーネルのPythonラッパーを含んだ、Pythonモジュールを返します。
+たとえば、オペレーションを1度ビルドしたら、Pythonから次のように実行できます。
+
+```python
+import tensorflow as tf
+zero_out_module = tf.load_op_library('./zero_out.so')
+with tf.Session(''):
+  zero_out_module.zero_out([[1, 2], [3, 4]]).eval()
+
+# Prints
```

Comment:
[weak suggestion] Do we need to translate this, too?

Reply:
Thanks. Translated it.